geospaas_processing/downloaders.py

```diff
@@ -251,14 +262,7 @@ def get_provider_settings(self, url_prefix):
                 return self.provider_settings[prefix]
         return {}
 
-    def find_dataset_file(self, file_prefix):
-        """Find a downloaded file for the dataset"""
-        for filename in os.listdir(self.download_directory):
-            if re.match(f"^{file_prefix}(|_.*)$", filename):
-                return filename
-        return None
-
-    def download_dataset(self, dataset):
+    def download_dataset(self, dataset, download_directory):
```

Comment:
I have chaned the name of self.download_directory attribute. We need that name in the argument because we have lots of other arguments and this name help us not to confuse.

Done in the next commit.

Reply:
Instead of changing the name of an existing instance attribute, please change the name of the parameter that you are adding.

Also, please use explicit names, and as few abbreviations as possible in variable names.
In this case you can use `download_folder` instead of `d_directory`.